src/scenes/Projects/List/ProjectList.tsx

```diff
@@ -64,14 +86,24 @@ export const ProjectList: FC = () => {
           <Skeleton width="12ch" />
         )}
       </Typography>
-      <ListContainer>
-        {listOrPlaceholders(data?.projects.items, 5).map((item, index) => (
-          <ProjectListItemCard
-            key={item?.id ?? index}
-            project={item}
-            className={classes.projectItem}
-          />
-        ))}
+      <ListContainer className={classes.listContainer}>
+        <VirtualList
+          dataLength={currentItemsCount}
+          next={loadMoreItems}
+          hasMore={hasMore}
+          itemType="project"
+          setContainerHeight={setContainerHeight}
+        >
+          {listOrPlaceholders(data?.projects.items, itemsPerPage).map(
+            (item, index) => (
+              <ProjectListItemCard
```

Comment:
Could `VirtualList` use a render prop (or children function) to render individual rows? That would allow rendering to be decoupled from list pagination logic.

Reply:
I mean, I'm sure it could. I'm struggling to understand the benefit, though... maybe because it's getting late and I'm tired. The consuming component still has to pass in basically all the same information, except with a layer of abstraction now inserted. To me, render props are for when you need more than one "children" slot, or when you have to do some kind of extra logic to the `children` before you render them. Neither seems to really be the case here.